### PR TITLE
Changed Client to use 1.1 and more unit tests

### PR DESCRIPTION
**Pull request by hc-jordenlowe**
Should be 4 unit tests:

1. One for happy path, parses an API User
2. One to handle client errors.
3. One to handle server errors (like say if they are not of a json form)
4. One to handle null access tokens.

This also changed up the client for the better.  On top of that moved all projects to be .net 1.x

### CODE REVIEW

**Comment by hc-jordenlowe** [comment]
Just added nuspec, this will be used when we create the nuget pack in the CI pipeline

File: Fabric.Authorization.Client/scripts/Fabric.Authorization.Client.nuspec [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (michaelvidal24) [inline Fabric.Authorization.Client/scripts/Fabric.Authorization.Client.nuspec]: Did we end up needing a nuspec file? I thought I remembered something about .net standard project using the .csproj file to figure out what needs to be packed. I could be wrong though.

File: Fabric.Authorization.Client.UnitTests/AuthorizationClientTests.cs [hunk 2]
```diff
@@ -1,35 +1,145 @@
+using System;
```
Comment (bsmithvt) [inline Fabric.Authorization.Client.UnitTests/AuthorizationClientTests.cs]: What is everyone's thoughts on establishing a naming convention for unit/integration test method names? Below is the convention I've used so far for Identity/Auth (and other companies in the past) that seems to work well:

http://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html

Comment (hc-jordenlowe) [reply Fabric.Authorization.Client.UnitTests/AuthorizationClientTests.cs]: I can rename these to something of the nature:

<Function name>_<What we are testing>_<Expected result>

That naming convention works fine, but may be long.  Depends on the test runner used, if you need all of that information.  But I am fine with that format.

File: Fabric.Authorization.Client/AuthorizationClient.cs [hunk 3]
```diff
@@ -15,13 +16,29 @@ public AuthorizationClient(HttpClient client)
             this.client = client;
         }
 
-        public async Task<UserApiModel> GetPermissionsForCurrentUser()
+        public async Task<UserApiModel> GetPermissionsForCurrentUser(string accessToken)
         {
+            CheckAccessToken(accessToken);
+
```
Comment (michaelvidal24) [inline Fabric.Authorization.Client/AuthorizationClient.cs]: I missed this on the original code review, but does HttpRequestMessage need to be wrapping in a using? I think it implements IDisposable.

**Pull request closed without merging**